### PR TITLE
Feature/availability subscriber

### PR DESCRIPTION
**Pull request by jsjoaorsouza**
#### What problem is this solving?

This solution aims to solve the impossibility of changing the name and email placeholder text values in the availability subscriber block

#### How to test it?

Click on the link below to see the implementation. The values of the name and email placeholder are set by the block properties "namePlaceholder" and "emailPlaceholder". The block documentation was updated with the descriptions of the properties added.

https://avtjoao--avantivtexio.myvtex.com/variacao-tamanho/p?skuId=205

#### Screenshots or example usage:

image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +19,9 @@
       "eslint --fix",
       "prettier --write"
     ],
+    "*.{json,graphql,gql}": [
+      "prettier --write"
+    ],
     "*.json": [
       "prettier --write"
     ]
```
Comment (igorbrasileiro) [inline package.json:27]: Can you remove this?

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]
Hey @jsjoaorsouza why not change the message content? Has the same semantic. @victorhmp @icazevedo

File: package.json (hunk 1 above)
Comment (jsjoaorsouza) [reply package.json:27]: Hi igorbrasileiro! Thanks for your attention!

> Hey @jsjoaorsouza why not change the message content? Has the same semantic. @victorhmp @icazevedo

You mean, using this doc? https://developers.vtex.com/vtex-developer-docs/docs/storefront-content-internationalization

if yes, i was thinking in a solution where theres no need for graphql, but i will look into it! Thank you very much!

File: .eslintrc [hunk 2]
```diff
@@ -1,4 +1,7 @@
 {
   "extends": "vtex",
-  "root": true
-}
+  "root": true,
+  "env": {
+    "node": true
+  }
+}
```
Comment (victorhmp) [inline .eslintrc:7]: Why was this change necessary? 🤔

File: docs/AvailabilitySubscriber.md [hunk 3]
```diff
@@ -48,6 +48,12 @@ When implementing this component as a block, various inner blocks may be availab
   }
 ```
 
+| Prop name            | Type      | Description                                                                      | Default value      |
+| -------------------- | --------- | -------------------------------------------------------------------------------- | ------------------ |
+| `namePlaceholder`      | `string` | text value of the name input placeholder.  |  undefined         |
```
Comment (victorhmp) [inline docs/AvailabilitySubscriber.md:53]:
```suggestion
| `namePlaceholder`      | `string` | Text value of the name input placeholder.  |  `undefined`         |
```

Comment (victorhmp) [reply docs/AvailabilitySubscriber.md:53]: Wait, the default value is not `undefined`, right? The default value here should be the default placeholder displayed by the component today

File: docs/AvailabilitySubscriber.md [hunk 4]
```diff
@@ -48,6 +48,12 @@ When implementing this component as a block, various inner blocks may be availab
...
+| `emailPlaceholder`     | `string` | text value of the email input placeholder. |  undefined         |
```
Comment (victorhmp) [inline docs/AvailabilitySubscriber.md:54]: Same here

File: package.json [hunk 5]
```diff
@@ -1,5 +1,4 @@
 {
-  "version": "3.47.3",
   "name": "store-components",
```
Comment (victorhmp) [inline package.json:2]: Why was the version moved?

Comment (victorhmp) [reply package.json:2]: I think you can just revert all changes to this file, `.eslintrc` and `prettierrc`

File: react/AvailabilitySubscriber.tsx [hunk 6]
```diff
@@ -160,9 +163,13 @@ function AvailabilitySubscriber(props: Props) {
...
               type="text"
-              placeholder={intl.formatMessage({
-                id: 'store/availability-subscriber.name-placeholder',
-              })}
+              placeholder={
+                props.namePlaceholder
+                  ? props.namePlaceholder
+                  : intl.formatMessage({
+                      id: 'store/availability-subscriber.name-placeholder',
+                    })
```
Comment (victorhmp) [inline react/AvailabilitySubscriber.tsx:171]: Move this to a variable outside the markup. And this conditional can also be simplified to something like:

```js
props.namePlaceholder || intl.formatMessage({...})
```

File: react/AvailabilitySubscriber.tsx [hunk 7]
```diff
@@ -171,9 +178,13 @@ function AvailabilitySubscriber(props: Props) {
             <Input
               name="email"
               type="text"
-              placeholder={intl.formatMessage({
-                id: 'store/availability-subscriber.email-placeholder',
-              })}
+              placeholder={
+                props.emailPlaceholder
```
Comment (victorhmp) [inline react/AvailabilitySubscriber.tsx:182]: Same comment from above

**Review by victorhmp — CHANGES_REQUESTED** [review]

**Comment by victorhmp** [comment]
Closing this since it's stale.
